simplex_transforms/stan/transforms/Exponential/Exponential_functions.stan

```diff
@@ -0,0 +1,21 @@
+vector exponential_simplex_constrain_lp(vector y) {
+  int N = rows(y);
+  vector[N] x_pos = y - log1p_exp(-y);
+  real r = log_sum_exp(x_pos);
+  vector[N] x = exp(x_pos - r);
+  target += x_pos;
+  target += std_normal_lpdf(r - log(N));
+  target += sum(y) - N * r;
+  return x;
+}
+
+vector exponential_log_simplex_constrain_lp(vector y) {
+  int N = rows(y);
+  vector[N] x = y - log1p_exp(-y);
+  real r = log_sum_exp(x);
+  vector[N] log_x = x - r;
+  target += x;
+  target += std_normal_lpdf(r - log(N));
+  target += log_x[N];
+  return log_x;
```

Comment:
Same here, I can return `x - r` instead of allocating but will need to make the last term `log_x[N]`

Reply:
Like above, I prefer to explicitly define `log_x`.